### DISCUSSION TITLE
[Feature] Allow padding for text

### DISCUSSION THREAD

**Discussion by Eschguy** [damongolding/immich-kiosk#241] [Feature request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Allow, maybe in scaling or pixel count, a way to bring info text (date/time/weather) in a bit. I'm repurposing an digital frame with an old tablet, and it's juuuuuuust getting cut off. 

I saw a discussion about the size of text, but not the padding from the edge.

![PXL_20241219_170931519](https://github.com/user-attachments/assets/932bdc8c-b5bc-4da1-8dd9-18e47bae44da)

**Comment by damongolding** [comment]
You can fix that with custom css. I think an option to adjust that might be too granular. 

I can have a look at what css you could add to adjust it when I'm back behind a computer if you're not familiar with css styling.

**Reply by Eschguy** [reply]
That's fair. I'm useless when it comes to css, so I was hoping for a more config-level approach to help others as well.

**Comment by damongolding** [comment]
Your setup looks pretty sweet btw!

**Reply by Eschguy** [reply]
Thanks! The old tech part crapped out, but I had a spare tablet lying around so I figured just kinda.....smush 'em.

**Comment by damongolding** [comment]
All Kiosks elements use the font as a base, so if you bump up the `font_base` the padding, as well as the font size increases.

That aside the below will push the clock, weather and metadata in proportionally. 

I've used rem here but you can change it px. 1 rem = 16px, 1.5rem = 24px, 2rem = 32px etc

[custom css docs](https://github.com/damongolding/immich-kiosk?tab=readme-ov-file#custom-css)


```css
#clock,
#weather,
.image--metadata {
    padding: 2rem; 
}
```

**Reply by Eschguy** [reply]
That did the trick, I appreciate the help.

**Comment by Eschguy** [comment]
Sorry to necro this discussion, but CSS appears to not affect the time or the names/ages of people in the photo. Has there been any further consideration given to the idea of just a global padding feature?

**Comment by damongolding** [comment]
@Eschguy I added `frame_padding` in this beta build. Want to give it a try? `hcr.io/damongolding/immich-kiosk-development:0.25.0-beta.1`

examples:

```yaml
frame_padding: 25 # this would give Kiosk 25px around all side
```

```yaml
frame_padding: 25  50 # this would give Kiosk 25px top and bottom and 50px left and right
```

```yaml
frame_padding: 20  30 40 50  # this would give Kiosk 20px top, 30px right, 40px bottom and 50 px left
```

**Reply by Eschguy** [reply]
The clock seems padded correctly, but the name/date block and weather block don't seem to be affected. Tried a variety of the three options in my `config.yaml`

Does `custom_css` need to remain true?

**Reply by damongolding** [reply]
I added a couple of fixes, once should solve this issue. New image is `ghcr.io/damongolding/immich-kiosk-development:0.25.0-beta.2`

**Reply by damongolding** [reply]
you should be able to remove the custom css you were using if you like

**Reply by Eschguy** [reply]
It works! Just a few notes:

- It pads the photo and the text, not just text. Depending on the photo, squishing occurs
- `custom.css` needs to be removed, as it is additive. Took a minute to figure out why the clock was so far in
- For documentation purposes, the multiple padding preferences are int arrays, so `[x, y, z, a]` is needed, not just `x y z a`

**Reply by Eschguy** [reply]
After playing around a bit, was able to get what I liked. I would think at some point it would be ideal that it could also be defined in the URL, in the instance of multiple kiosk frames. 
![PXL_20250922_222825215](https://github.com/user-attachments/assets/ff75a571-3209-40ae-b6f2-08d9278d8f49)

**Reply by damongolding** [reply]
You can define the padding in the  url. 

All:
```url
http://URL?frame_padding=20
```

(Top and bottom) and (left and right):
```url
http://URL?frame_padding=20&frame_padding=50
```

**Reply by Eschguy** [reply]
Fantastic, appreciate the work on this!
